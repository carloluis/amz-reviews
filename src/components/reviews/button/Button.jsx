import React from 'react';
import PropTypes from 'prop-types';
import styles from './Button.scss';

const Button = ({ onClick, text }) => (
    <a className={styles.button} onClick={onClick}>
        {text}
    </a>
);

Button.propTypes = {
    onClick: PropTypes.func.isRequired,
    text: PropTypes.node.isRequired
};

export default Button;
