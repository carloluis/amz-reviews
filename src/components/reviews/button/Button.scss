@import '../../../styles/palette.scss';

.button {
    display: inline-block;
    background: $dark-grey;
    color: $white;
    width: 160px;
    height: 32px;
    border-radius: 16px;
    text-align: center;
    text-transform: uppercase;
    line-height: 32px;
    font-size: 14px;
    font-weight: 600;
    cursor: pointer;
    transition: all 0.2s ease-in-out;

    &:hover {
        background: $dark-blue;
    }
}
