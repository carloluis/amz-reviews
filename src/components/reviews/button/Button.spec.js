import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import Button from './Button';

describe('<Button/>', () => {
    it('snapshot', () => {
        const result = renderer.create(<Button text="Some Text" onClick={jest.fn()} />);
        expect(result).toMatchSnapshot();
    });
});

describe('Button', () => {
    it('should call the callback when click', () => {
        const onClickMock = jest.fn();
        const wrapper = shallow(<Button text="" onClick={onClickMock} />);
        wrapper.simulate('click');
        expect(onClickMock).toBeCalled();
    });
});
