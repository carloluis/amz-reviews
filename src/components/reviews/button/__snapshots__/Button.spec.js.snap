// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button/> snapshot 1`] = `
<a
  className="button"
  onClick={[MockFunction]}
>
  Some Text
</a>
`;
